packages/Avalonia/AvaloniaBuildTasks.targets

```diff
@@ -42,12 +42,20 @@
   </Target>
   
   <PropertyGroup>
-    <BuildAvaloniaResourcesDependsOn>$(BuildAvaloniaResourcesDependsOn);AddAvaloniaResources;ResolveReferences</BuildAvaloniaResourcesDependsOn>
+    <BuildAvaloniaResourcesDependsOn>$(BuildAvaloniaResourcesDependsOn);AddAvaloniaResources;ResolveReferences;ReGenerateAvaloniaResourcesOnResourceDeletion</BuildAvaloniaResourcesDependsOn>
   </PropertyGroup>
   
+  <Target Name="ReGenerateAvaloniaResourcesOnResourceDeletion" BeforeTargets="GenerateAvaloniaResources">
+    <Hash ItemsToHash="@(AvaloniaResource);@(AvaloniaXaml);$(MSBuildAllProjects)">
+      <Output TaskParameter="HashResult" PropertyName="UpdatedInputCacheContents" />
```

Comment:
Use `AvaloniaResourcesDependencyHash`. It's a good idea to prefix properties like that to avoid name clashes with other nuget packages